processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -0,0 +1,161 @@
+import io.druid.segment.column.ColumnCapabilitiesImpl;
+import io.druid.segment.column.ValueType;
+
+import java.util.List;
+
+public final class DimensionHandlerUtils
+{
+  private DimensionHandlerUtils() {}
+
+  public final static ColumnCapabilities DEFAULT_STRING_CAPABILITIES =
+      new ColumnCapabilitiesImpl().setType(ValueType.STRING)
+                                  .setDictionaryEncoded(true)
+                                  .setHasBitmapIndexes(true);
+
+  public static DimensionHandler getHandlerFromCapabilities(
+      String dimensionName,
+      ColumnCapabilities capabilities,
+      MultiValueHandling multiValueHandling
+  )
+  {
+    if (capabilities == null) {
+      return new StringDimensionHandler(dimensionName, multiValueHandling);
+    }
+
+    if (dimensionName.equals(Column.TIME_COLUMN_NAME)) {
+      return new StringDimensionHandler(Column.TIME_COLUMN_NAME, MultiValueHandling.ARRAY);
+    }
+
+    multiValueHandling = multiValueHandling == null ? MultiValueHandling.ofDefault() : multiValueHandling;
+
+    if (capabilities.getType() == ValueType.STRING) {
+      if (!capabilities.isDictionaryEncoded() || !capabilities.hasBitmapIndexes()) {
+        throw new IAE("String column must have dictionary encoding and bitmap index.");
+      }
+      return new StringDimensionHandler(dimensionName, multiValueHandling);
+    }
+
+    // Return a StringDimensionHandler by default (null columns will be treated as String typed)
+    return new StringDimensionHandler(dimensionName, multiValueHandling);
+  }
+
+  public static <ColumnSelectorStrategyClass extends ColumnSelectorStrategy> ColumnSelectorStrategyClass makeStrategy(
+      ColumnSelectorStrategyFactory<ColumnSelectorStrategyClass> strategyFactory,
+      String dimName,
+      ColumnCapabilities capabilities,
+      List<String> availableDimensions
+  )
+  {
+    capabilities = getEffectiveCapabilities(dimName, capabilities, availableDimensions);
+    return strategyFactory.makeColumnSelectorStrategy(dimName, capabilities);
+  }
+
+  public static <ColumnSelectorStrategyClass extends ColumnSelectorStrategy> ColumnSelectorPlus<ColumnSelectorStrategyClass>[] getDimensionInfo(
+      ColumnSelectorStrategyFactory<ColumnSelectorStrategyClass> strategyFactory,
+      List<DimensionSpec> dimensionSpecs,
+      StorageAdapter adapter,
+      ColumnSelectorFactory cursor
+  )
+  {
+    int dimCount = dimensionSpecs.size();
+    ColumnSelectorPlus<ColumnSelectorStrategyClass>[] dims = new ColumnSelectorPlus[dimCount];
+    for (int i = 0; i < dimCount; i++) {
+      final DimensionSpec dimSpec = dimensionSpecs.get(i);
+      final String dimName = dimSpec.getDimension();
+      ColumnSelectorStrategyClass strategy = makeStrategy(
+          strategyFactory,
+          dimName,
+          cursor.getColumnCapabilities(dimSpec.getDimension()),
+          adapter == null ? null : Lists.newArrayList(adapter.getAvailableDimensions())
```

Comment:
Hm, good point, I had it originally to check for metrics but that's irrelevant with the later non-String type check.

I've removed that and the storage adapter from the params